javalin/src/main/java/io/javalin/config/RoutingConfig.kt

```diff
@@ -11,4 +11,7 @@ class RoutingConfig {
     @JvmField
     var treatMultipleSlashesAsSingleSlash = false
 
+    @JvmField
+    var caseInsensitiveRoutes = false
```

Comment:
The one thing I am worried about is how this change would work together with config.plugins.enableRedirectToLowercasePaths();. Can you add a test for this interaction?

I guess we could always throw an exception when these two are combined together.

Reply:
Throws `java.lang.RuntimeException: java.lang.IllegalArgumentException: Paths must be lowercase when using RedirectToLowercasePathPlugin`

Reply:
But what if somebody does something like this:

```kotlin
val app = Javalin.create { config ->
  config.plugins.enableRedirectToLowercasePaths()
  it.routing.caseInsensitiveRoutes = true
}.get("/hello") { ctx ->
  ctx.result("hi")
}.start()
// Client
http.get("/HELLO")
```

We should ideally throw an exception before the server gets up and running.